cmd/skaffold/app/cmd/build_test.go

```diff
@@ -87,22 +113,24 @@ func TestQuietFlag(t *testing.T) {
 }
 
 func TestRunBuild(t *testing.T) {
-	errRunner := func(context.Context, io.Writer) ([]build.Artifact, error) {
+	errRunner := func(context.Context, io.Writer) ([]build.Result, error) {
 		return nil, errors.New("some error")
 	}
-	mockCreateRunner := func(context.Context, io.Writer) ([]build.Artifact, error) {
-		return []build.Artifact{{
-			ImageName: "gcr.io/skaffold/example",
-			Tag:       "test",
+	mockCreateRunner := func(context.Context, io.Writer) ([]build.Result, error) {
+		return []build.Result{{
+			Result: build.Artifact{
+				ImageName: "gcr.io/skaffold/example",
+				Tag:       "test",
+			},
 		}}, nil
 	}
-	defer func(f func(context.Context, io.Writer) ([]build.Artifact, error)) {
+	defer func(f func(context.Context, io.Writer) ([]build.Result, error)) {
```

Comment:
I know this is not introduced by your change, but just noticed that this deferred re-establishing the original `createRunnerAndBuildFunc` should be in the above test as well, in `TestQuietFlag`.

Reply:
i think it could be me who added this file. I wanted to add tests for deploy but then it slipped. 
https://github.com/GoogleContainerTools/skaffold/pull/2085